core/testing.go

```diff
@@ -40,9 +42,9 @@ func DefaultTestConfig() *testnode.Config {
 	tmConfig.Consensus.TimeoutCommit = time.Millisecond * 200
 
 	return testnode.DefaultConfig().
-		WithChainID(chainID).
-		WithFundedAccounts(generateRandomAccounts(10)...). // 10 usually is enough for testing
 		WithGenesis(genesis).
+		WithFundedAccounts(generateRandomAccounts(10)...). // 10 usually is enough for testing
+		WithChainID(chainID).
```

Comment:
this is a super subtle change, the chainID field is stored on the genesis type, so in the old version of the code, it set a chainID on a genesis, the overwrote the genesis. Internally the chainID is used in two different places now and so it must be propagated correctly by assigning the genesis, then modifying the chainID if required.